### PR TITLE
Improve the rendering of mempool trace messages

### PR DESCRIPTION
**Pull request by dcoutts**
Crucially, include the `ApplyTxErr` rejection reason in the `TraceMempoolRejectedTx` case.

### CODE REVIEW

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs [hunk 1]
```diff
@@ -1201,32 +1202,40 @@ instance Condense (HeaderHash blk)
...
-instance (Show (GenTx blk), Show (GenTxId blk))
+instance (Condense (GenTx blk), Condense (GenTxId blk), Show (ApplyTxErr blk))
  => ToObject (TraceEventMempool blk) where
-  toObject _verb (TraceMempoolAddedTx tx _mpSzBefore mpSzAfter) =
+  toObject verb (TraceMempoolAddedTx tx _mpSzBefore mpSzAfter) =
     mkObject
       [ "kind" .= String "TraceMempoolAddedTx"
-      , "txAdded" .= String (pack $ show tx)
-      , "mempoolSize" .= String (pack $ show mpSzAfter)
+      , "tx" .= condense tx
```
Comment (Jimbo4350) [inline cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: Should we avoid using `condense` as per https://github.com/input-output-hk/cardano-node/issues/721#issuecomment-604343274

Comment (CodiePP) [reply cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: yes, condense should be banned.

Comment (dcoutts) [reply cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: Ok, then we should use `ToObject` and provide an instance for it, and add it to `TraceConstraints`.

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs [hunk 2]
```diff
@@ -1201,32 +1202,40 @@ instance Condense (HeaderHash blk)
...
       [ "kind" .= String "TraceMempoolManuallyRemovedTxs"
-      , "txsManuallyRemoved" .= String (pack $ show txs0)
-      , "txsNoLongerValidRemoved" .= String (pack $ show txs1)
-      , "mempoolSize" .= String (pack $ show mpSz)
+      , "txsRemoved" .= map condense txs0
+      , "txsInvalidated" .= map condense txs1
+      , "mempoolSize" .= toObject verb mpSz
+      ]
+
+instance ToObject MempoolSize where
```
Comment (CodiePP) [inline cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: :+1:

**Pull request merged**